src/plugins/intel_gpu/src/graph/include/kernel_selector_helper.h

```diff
@@ -141,12 +142,16 @@ struct kernel_impl_params {
                        , prog(_prog)
                        , desc(_desc)
                        , unique_id(_uid)
-                       , input_layouts(_int_layouts)
+                       , input_layouts(_in_layouts)
                        , output_layout(_out_layout)
                        , fused_desc(_fused_descs)
                        , fused_act_funcs(_fused_act_funcs)
                        , activation_params(_act_params)
-                       , primary_input_idx(0) {}
+                       , primary_input_idx(0) {
+        if (!_in_layouts.empty() && !_in_layouts[0].get_partial_shape().rank().is_dynamic()) {
+            ov_input_rank = _in_layouts[0].get_shape().size();
```

Comment:
Again, why do we need that as common field?
To me it looks like all you need is to modify gather_nonzero_impl::create and count_nonzero_impl::create functions and kernel parameters for those 2 operations to pass original rank that may be taken in the moment of conversion from impl_params.input_layouts.

Reply:
I agree, I fixed. Thanks.